main.cpp

```diff
@@ -75,29 +80,48 @@ struct <#structName1#>                                //4
 
 struct U
 {
-    float <#name1#> { 0 }, <#name2#> { 0 };
-    <#returnType#> <#memberFunction#>(<#type name#>* <#updatedValue#>)      //12
+    float uObject1 { 10.f }, uObject2 { 0.01f };
+    
+    float uDoAThing(float* updatedUObject)      //12
     {
-        
+        if(updatedUObject != nullptr)
+        {
+            std::cout << "U's uObject1 value: " << this->uObject1 << std::endl;
+            this->uObject1 = *updatedUObject;
+            std::cout << "U's uObject1 updated value: " << this->uObject1 << std::endl;
+            while( std::abs(this->uObject2 - this->uObject1) > 0.001f )
+            {
+                ++this->uObject2;
+                --this->uObject1;
+            }
+            
+        }
+        std::cout << "U's uObject2 updated value: " << this->uObject2 << std::endl;
+        return this->uObject2 * this->uObject1;
     }
 };
 
-struct <#structname2#>
+struct FourthClass
 {
-    static <#returntype#> <#staticFunctionA#>(U* that, <#type name#>* <#updatedValue#> )        //10
+    static float staticDoAThing(U* that, float* valueUpdated)        //10
     {
-        std::cout << "U's <#name1#> value: " << that-><#name1#> << std::endl;
-        that-><#name1#> = <#updatedValue#>;
-        std::cout << "U's <#name1#> updated value: " << that-><#name1#> << std::endl;
-        while( std::abs(that-><#name2#> - that-><#name1#>) > 0.001f )
+        if(that && valueUpdated != nullptr)
```

Comment:
`&&` does not work like this.
You need to compare each variable with `nullptr` separately. 
```
if( that != nullptr && valueUpdated != nullptr )
```